packages/flutter/lib/src/widgets/scrollable.dart

```diff
@@ -19,6 +19,24 @@ import 'page_storage.dart';
 import 'scroll_behavior.dart';
 import 'scroll_configuration.dart';
 
+/// Identifies one or both limits of a [Scrollable] in terms of its scrollDirection.
+enum ScrollableEdge {
+  /// The top and bottom of the scrollable if its scrollDirection is vertical
+  /// or the left and right if its scrollDirection is horizontal.
+  both,
+
+  /// Only the top of the scrollable if its scrollDirection is vertical,
+  /// or only the left if its scrollDirection is horizontal.
+  leading,
```

Comment:
We do use start/end in the same sense for ViewportAnchor.  However edges are usually characterized as leading and trailing. I could change the enum to ScrollableLimit with values start, end, both, none?